src/libmeasurement_kit/ooni/web_connectivity.cpp

```diff
@@ -524,8 +530,14 @@ static void experiment_dns_query(
     SharedPtr<Reactor> reactor, SharedPtr<Logger> logger) {
 
     if (net::is_ip_addr(hostname)) {
-        // Don't perform DNS resolutions if it's an IP address
-        // XXX This means we are not filling the entry
+        // Don't perform DNS resolutions if it's an IP address. The spec of
+        // web connectivity does not say what to do in case we are not doing
+        // the DNS step. A good value to indicate we've not done this step
+        // seems to be `null`. That is also Go-friendly because you can use
+        // a pointer in the structure definition.
+        //
+        // See <https://github.com/ooni/spec/issues/148>.
+        (*entry)["queries"] = nullptr;
```

Comment:
Can we initialise this to an empty list instead of a null value?

Reply:
We can. Why do you think an empty list is better than a `null` value? My reasoning for using `null` is that it indicates the field has not / could not been set. This fits more nicely with the semantic of an optional field than leaving the list empty, I believe.

Reply:
I was thinking of it from the perspective of a consumer of the data as it's easier to have it be of the same type than having to check for null values.

I see your point though and what you propose is possibly more consistent with the approach we currently following WRT to data formats.

I am now convinced that using `null` here is the right thing to do.

Reply:
Awesome, so I don't think I need to do anything special here because that's what we already have written in https://github.com/measurement-kit/measurement-kit/issues/1866

Reply:
Well, except perhaps opening a PR to address https://github.com/ooni/spec/issues/148